### DISCUSSION TITLE
Link email provide after login with Github

### DISCUSSION THREAD

**Discussion by rebaz94** [supabase/supabase#6811] [Questions] [answered]
Hi
I want to link logged user with email after signing up from external OAuth provider.
for example this create new user account
```dart
await supabase.client.auth.signInWithProvider(Provider.github);
```

according to the doc using same email will link to existing account
```dart
final res = await supabase.client.auth.signUp(
   'rebazrauf@gmail.com',
   'rebaz12345678',
);
print(res.error);
print(res.user?.id)
```

but after doing this, anytime this code executed it will create new user but if you check ```users``` table the row for that user id not exist!

after checking the log it say it repeated signup but it does not link to existing account and also return fake user which not exist.

here the log
```
{"action":"user_repeated_signup","actor_id":"f509ed4d-3842-47ee-bbe7-56e258bbb92a","actor_name":"Rebaz Raouf","actor_username":"rebazrauf@gmail.com","log_type":"user","timestamp":"2022-05-09T10:15:19Z","traits":{"provider":"email"}}
```

if you try login with same info it will say invalid credential.

Please could anyone explain this is a bug or how works

**Comment by GaryAustin1** [comment, marked as answer]
Some limited testing here. https://github.com/supabase/supabase/discussions/5827#discussioncomment-2305049
 Had to use update password to get email login working after 3rd party.

**Reply by rebaz94** [reply]
Thank you for the response. I will try and let you know if thats working

**Reply by rebaz94** [reply]
I tried to update password after login with 3rd party and it works. but email provider is not in provider list,  maybe cause problem in future? 
```{"provider":"github","providers":["github"]}```

but if you signup using email and then connect with 3rd party provider it works and add provider to the list
```{"provider":"email","providers":["email","github"]}```

also that if you call ```signUp``` with same email and password it return fake user.

**Reply by GaryAustin1** [reply]
Yes, your observations on order of email are the same as I posted in my testing.
I also reached out to SB staff in that post, hoping for confirmation multiple login methods per email were supported, but got no answer.
